### PR TITLE
Handle Lua script loading errors

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
Handle errors when loading Lua scripts.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by omoerbeek — COMMENTED** [review]
This looks good to me, but @Habbie needs to review too, as this is a change in behaviour wrt the lua backend: previously only an error was logged, and now it throws if the Lua file cannot be found or is not accessible.

File: pdns/lua-base4.cc [hunk 1]
```diff
@@ -17,18 +18,18 @@
...
+void BaseLua4::loadFile(const std::string& fname)
 {
-  int ret = 0;
   std::ifstream ifs(fname);
   if (!ifs) {
-    ret = errno;
-    SLOG(g_log << Logger::Error << "Unable to read configuration file from '" << fname << "': " << stringerror(ret) << endl,
-         g_slog->withName("lua")->error(Logr::Error, ret, "Unable to read configuration file", "file", Logging::Loggable(fname)));
-    return ret;
+    assert(errno != 0);
```
Comment (Habbie) [inline pdns/lua-base4.cc:25]: does this assert have a risk of stopping the whole process?

Comment (Habbie) [reply pdns/lua-base4.cc:25]: or is the point that a failing ifstream that leaves errno zero is a very clear bug, in which case exiting is fine?

**Review by Habbie — COMMENTED** [review]
One question, looks good otherwise!

File: pdns/lua-base4.cc (hunk 1 above)
Comment (fredmorcos) [reply pdns/lua-base4.cc:25]: It's the latter, yes.

Comment (omoerbeek) [reply pdns/lua-base4.cc:25]: That's what I thought as well. Though on second thought: while system calls are guaranteed to set `errno` on error, in general lib functions are not. So I prefer the assert to be removed.

Comment (fredmorcos) [reply pdns/lua-base4.cc]: Alright, done. I fixed-up and force-pushed, I hope that's OK since the change is simple enough.

Comment (Habbie) [reply pdns/lua-base4.cc]: :+1:

**Review by Habbie — APPROVED** [review]

**Pull request merged**